### PR TITLE
feat: Queue worker for order creation

### LINKED ISSUE #168: Create queue and worker for ACME order creation
- Use the Let's encrypt class to create a new order
- Add DNS entries (if DNS system is ready)
- Add data to Prisma

This fills out step 1 of
https://github.com/Seneca-CDOT/starchart/issues/167

### PR DESCRIPTION
**Pull request by dadolhay**
- Moves directory url to ENV
- Alters the Let's encrypt library to not always reload challengegs, we only need that when the order is created
- Alters Prisma schema (+ seed) to comply with our needs (modified Certificate from one-to-one => one-to-many because the foreign key constraint makes testing impossible otherwise)
- Updates FE to match new db schema
- Updates related models to better match our needs
- Implements worker to handle certificate creation flow

Closes #168

### CODE REVIEW

File: app/components/certificate/description.tsx [hunk 1]
```diff
@@ -21,14 +21,18 @@ export default function DescriptionSection({
       <Text>{description}</Text>
       {certRequested && (
```
Comment (Ririio) [inline app/components/certificate/description.tsx:22]: I don't think we need `certRequested` if we're going to validate based off `validFrom` and `validTo`. We can move the validation above to start from `<HStack>`

File: app/routes/__index/certificate/index.tsx [hunk 2]
```diff
@@ -16,7 +16,8 @@ export default function CertificateIndexRoute() {
   const certificate: Certificate = {
     id: 1,
     username: 'user_test',
-    subject: `*.user_test.example.com`,
+    domain: `user_test.example.com`,
```
Comment (Ririio) [inline app/routes/__index/certificate/index.tsx:19]: This one actually confuses me a bit, from my understanding we'll have one certificate per user. If that was the case, do we have to display all available domains, or do the `*`

File: app/lib/lets-encrypt.server.ts [hunk 3]
```diff
@@ -69,9 +69,9 @@ const getAuthoritativeResolverForDomain = async (domain: string): Promise<Promis
 
   return resolver;
 };
-interface ChallengeBundle extends AcmeDnsChallenge {
+export interface ChallengeBundle extends AcmeDnsChallenge {
   domain: string;
-  value: String;
+  value: string;
```
[the remark this answers is not in the document]
Comment (humphd) [reply app/lib/lets-encrypt.server.ts:74]: It looks like https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-types.md would do this.  We should consider adding it in a follow-up.

**Review by humphd — COMMENTED** [review]
I was able to run this locally, and the steps I took where:

```sh
npx prisma db push --force-reset
npm run db:seed
```

It might be good to file an issue to add a `db:reset` script to `package.json` for the future.  I didn't know about being able to do this.

After the db was setup, I went to `http://localhost:8080/dev` and clicked Request Certificate, and the server logs show the flow happening.

I think this is looking good.  Made a few comments/questions.

File: app/models/certificate.server.ts [hunk 4]
```diff
@@ -4,61 +4,37 @@ import { prisma } from '~/db.server';
...
+      take: 1,
+    })
+    .then(([certificate]) => certificate);
+}
+
+export function getCertificateById(id: Certificate['id']) {
+  return prisma.certificate.findUnique({ where: { id } });
 }
 
 export async function createCertificate(
```
[the remark this answers is not in the document]
Comment (dadolhay) [reply app/models/certificate.server.ts:22]: Nice catch, it was an oversight

File: app/lib/lets-encrypt.server.ts [hunk 5]
```diff
@@ -173,24 +173,34 @@ class LetsEncrypt {
...
-
       this.#accountKey = LETS_ENCRYPT_ACCOUNT_PRIVATE_KEY_PEM;
-      this.#directoryUrl = acme.directory.letsencrypt.production;
     } else {
       // For testing and local development, let's use an ad-hoc generated key
 
       this.#accountKey = (await acme.crypto.createPrivateKey()).toString();
-      this.#directoryUrl = 'https://127.0.0.1:14000/dir';
 
       /**
```
[the remark this answers is not in the document]
Comment (dadolhay) [reply app/lib/lets-encrypt.server.ts:198]: Possibly, we will have to address those if they come up

**Review by humphd — APPROVED** [review]
Nice

**Review by cychu42 — APPROVED** [review]
LGTM. I have no further comment. Nice work!

**Pull request merged**